### PR TITLE
[Paddle-TRT] NMS Performance Optimization

### PR DESCRIPTION
**Pull request by zlsh80826**
### PR types

Performance optimization

### PR changes

OPs

### Describe

1. Replace mask vector to raw ptr in NMS:
Original implementation introduces an unnecessary memory copy of mask from host to device before launching NMSKernel
Origin:
image
This PR:
image

2. Launch NMSKernel on context stream to prevent implicit synchronization with default stream
3. The is ~1% e2e performance gain on trt-fp16/maskrcnn inference

### CODE REVIEW

**Review by cryoco — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/detection/bbox_util.cu.h [hunk 1]
```diff
@@ -276,14 +276,19 @@ static void NMS(const platform::CUDADeviceContext &ctx, const Tensor &proposals,
   const T *boxes = proposals.data<T>();
   auto place = BOOST_GET_CONST(platform::CUDAPlace, ctx.GetPlace());
   framework::Vector<uint64_t> mask(boxes_num * col_blocks);
```
Comment (jerrywgz) [inline paddle/fluid/operators/detection/bbox_util.cu.h]: mask could be removed

Comment (zlsh80826) [reply paddle/fluid/operators/detection/bbox_util.cu.h]: done

**Review by jerrywgz — APPROVED** [review]
LGTM

**Pull request merged**